packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -840,19 +948,15 @@ export class Summarizer extends EventEmitter implements ISummarizer {
     }
 
     /** Implementation of SummarizerInternalsProvider.generateSummary */
-    public async generateSummary(options: {
-        fullTree: boolean,
-        refreshLatestAck: boolean,
-        summaryLogger: ITelemetryLogger,
-    }): Promise<GenerateSummaryData | undefined> {
+    public async generateSummary(options: IGenerateSummaryOptions): Promise<GenerateSummaryData> {
+        const result = this.internalsProvider.generateSummary(options);
```

Comment:
Note: Moving this _before_ the check to stop for "parentNoLongerSummarizer" allows for the last summary attempt to still go through before stopping, and still getting the generate summary results back to caller.